packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -152,7 +152,7 @@ export class TableLayout<T> extends ListLayout<T> {
     this.setChildHeights(columns, height);
 
     rect.height = height;
-    rect.width = x;
+    rect.width = Math.max(this.virtualizer.visibleRect.width, x);
```

Comment:
hmm I don't think we had this before my changes right? So what's different now

Reply:
the header row now has overflow clip, previously it didn't
https://github.com/adobe/react-spectrum/blob/2cd8f0b509cd9a1d93322dd75fc1306e5f0e217f/packages/%40react-aria/virtualizer/src/VirtualizerItem.tsx#L79

at least, I thought it was that easy, however, now I'm looking back more... we were never setting allowsOverflow to true on the layout infos
Instead, in TableViewBase, we were directly overriding the style to overflow: visible

I've updated the PR for that method of solving it instead, we can choose whichever we like more